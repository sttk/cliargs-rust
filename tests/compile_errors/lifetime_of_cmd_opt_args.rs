use cliargs::Cmd;

fn main() {
    fn returns_one_of_opt_args() -> &'static str {
        let mut cmd =
            cliargs::Cmd::with_strings(["/path/to/app".to_string(), "--foo=bar".to_string()]);
        cmd.parse().unwrap();

        let opt_args = cmd.opt_args("foo").unwrap();
        println!("option arg (within the scope = {:?}", opt_args[0]);

        opt_args[0]
    }

    let opt_arg = returns_one_of_opt_args();
    println!("option arg (out of the scope) = {opt_arg:?}");
}
