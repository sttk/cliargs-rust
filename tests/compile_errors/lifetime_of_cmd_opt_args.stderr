warning: unused import: `cliargs::Cmd`
 --> tests/compile_errors/lifetime_of_cmd_opt_args.rs:1:5
  |
1 | use cliargs::Cmd;
  |     ^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0515]: cannot return value referencing local variable `cmd`
  --> tests/compile_errors/lifetime_of_cmd_opt_args.rs:12:9
   |
9  |         let opt_args = cmd.opt_args("foo").unwrap();
   |                        --- `cmd` is borrowed here
...
12 |         opt_args[0]
   |         ^^^^^^^^^^^ returns a value referencing data owned by the current function
